src/fictionalpancake/turbospork/paint/GraphicsHandler.java

```diff
@@ -0,0 +1,218 @@
+        g.drawText(style, fpsText, convertX(0), convertY(0));
+        lastPaint = System.currentTimeMillis();
+    }
+
+    private void drawNodeUnits(IPainter g, Node node) {
+        for (int owner : node.getUnitOwners()) {
+            int seed = getGroupSeed(node, owner);
+            drawUnitGroup(g, seed, seed, 1, owner, node.getUnits(owner), node.getX(), node.getY());
+        }
+    }
+
+    private void drawUnitGroup(IPainter g, UnitGroup group) {
+        int owner = group.getOwner();
+        drawUnitGroup(g, getGroupSeed(group.getSource(), owner), getGroupSeed(group.getDest(), owner), group.getProgress(), owner, group.getUnits(), group.getX(), group.getY());
+    }
+
+    public void checkScaling(IPainter g) {
+        int minDimension;
+        if (g.getWidth() < g.getHeight()) {
+            xOffset = 0;
+            yOffset = (g.getHeight() - g.getWidth()) / 2;
+            minDimension = g.getWidth();
+        } else {
+            xOffset = (g.getWidth() - g.getHeight()) / 2;
+            yOffset = 0;
+            minDimension = g.getHeight();
+        }
+        scale = ((double) minDimension) / GameConstants.FIELD_SIZE;
+    }
+
+
+    private void drawUnitGroup(IPainter g, int seed1, int seed2, double progress, int owner, int numUnits, double x, double y) {
+        Random rand1 = new Random(seed1);
+        Random rand2 = new Random(seed2);
+
+        int radius = convertSize(GameConstants.UNIT_RADIUS);
+        Color color = GameColors.getColorForOwner(owner).darker();
+        PaintStyle fillStyle = new PaintStyle();
+        fillStyle.color = color;
+        PaintStyle outlineStyle = new PaintStyle();
+        outlineStyle.color = Color.BLACK;
+        outlineStyle.fill = false;
+        for (int j = 0; j < numUnits; j++) {
+            Point coords1 = getRandomUnitCoords(rand1, x, y);
+            Point coords2 = getRandomUnitCoords(rand2, x, y);
+
+            int unitX = (int) (coords1.getX() + (coords2.getX() - coords1.getX()) * progress);
+            int unitY = (int) (coords1.getY() + (coords2.getY() - coords1.getY()) * progress);
+
+            // draw the unit
+            g.drawCircle(fillStyle, unitX, unitY, radius);
+            g.drawCircle(outlineStyle, unitX, unitY, radius);
+        }
+    }
+
+    private int getGroupSeed(Node node, int owner) {
+        return gameHandler.indexOf(node) * (owner + 1);
+    }
+
+
+    public void select(Node node) {
+        if (selectedNode == null && node != null && gameHandler.isInProgress()) {
+            int pos = gameHandler.getPosition();
+            if ((pos != -1 && pos == node.getOwner()) || node.getUnits(pos) > 0) {
+                selectedNode = node;
+                lastSelected = node;
+            }
+        }
+        else if(selectedNode != null && selectedNode == node) {
+            selectedNode = null;
+        }
+    }
+
+    public void attack(Node node) {
+        if (selectedNode != null && node != null && node != selectedNode) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l162)